crates/crypto/src/key_manager/local_key_manager.rs

```diff
@@ -0,0 +1,133 @@
+use crate::key::{KeyAlgorithm, PrivateKey, PublicKey};
+use crate::key_manager::key_store::{InMemoryKeyStore, KeyStore};
+use crate::key_manager::{KeyManager, KeyManagerError};
+use ssi_jwk::JWK;
+use std::sync::Arc;
+
+/// Implementation of the [`KeyManager`] trait with key generation local to the device/platform it
+/// is being run. Key storage is provided by a [`KeyStore`] trait implementation, allowing the keys
+/// to be stored wherever is most appropriate for the application.
+pub struct LocalKeyManager {
+    key_store: Arc<dyn KeyStore>,
+}
+
+impl LocalKeyManager {
+    /// Constructs a new `LocalKeyManager` that stores keys in the provided `KeyStore`.
+    pub fn new(key_store: Arc<dyn KeyStore>) -> Self {
+        Self { key_store }
+    }
+
+    /// Constructs a new `LocalKeyManager` that stores keys in memory.
+    pub fn new_in_memory() -> Self {
+        Self {
+            key_store: Arc::new(InMemoryKeyStore::new()),
+        }
+    }
+}
+
+impl KeyManager for LocalKeyManager {
+    fn generate_private_key(&self, key_algorithm: KeyAlgorithm) -> Result<String, KeyManagerError> {
+        let jwk = match key_algorithm {
+            KeyAlgorithm::Secp256k1 => JWK::generate_secp256k1(),
+            KeyAlgorithm::Secp256r1 => JWK::generate_p256(),
+            KeyAlgorithm::Ed25519 => JWK::generate_ed25519(),
+        }?;
+
+        let private_key = PrivateKey(jwk);
+        let public_key = private_key.to_public();
+        let key_alias = self.alias(&public_key)?;
+
+        self.key_store.insert(&key_alias, private_key)?;
+
+        Ok(key_alias)
+    }
+
+    fn get_public_key(&self, key_alias: &str) -> Result<Option<PublicKey>, KeyManagerError> {
+        if let Some(private_key) = self.key_store.get(key_alias)? {
+            Ok(Some(private_key.to_public()))
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn sign(&self, key_alias: &str, payload: &[u8]) -> Result<Vec<u8>, KeyManagerError> {
+        let private_key = self
+            .key_store
+            .get(key_alias)?
+            .ok_or(KeyManagerError::SigningKeyNotFound)?;
+
+        let signed_payload = private_key.sign(&payload.to_vec())?;
```

Comment:
is it assumed that all key types expose this sign method?

Reply:
That's correct. The `PrivateKey` struct is currently responsible for all key types, which itself is a thin wrapper around a JWK. 

Definitely flexible here to alternative approaches!

Reply:
I think it may be worth including a signing algorithm parameter, as sometimes the same key can be used with different algorithms (e.g. secp256k1 can be used with ECDH and ECDSA)

Reply:
This should still be possible, by the caller setting the algorithm field on the JWK. 

Currently, the algorithm is computed by this function in Spruce's `ssi-jwk` crate:
https://github.com/spruceid/ssi/blob/main/ssi-jwk/src/lib.rs#L341

This logic matches our signer lookup that's currently being done in Kotlin here:
https://github.com/TBD54566975/web5-kt/blob/main/crypto/src/main/kotlin/web5/sdk/crypto/Crypto.kt#L111C22-L111C22